taichi/transforms/type_check.cpp

```diff
@@ -274,7 +277,9 @@ class TypeCheck : public IRVisitor {
       error();
     }
     if (binary_is_bitwise(stmt->op_type)) {
-      if (!is_integral(stmt->lhs->ret_type.data_type)) {
+      auto lhs_dt = stmt->lhs->ret_type.data_type;
+      auto rhs_dt = stmt->rhs->ret_type.data_type;
+      if (!is_integral(lhs_dt) || !is_integral(rhs_dt)) {
```

Comment:
IIRC `type_check` guarantees that `BinaryOpStmt::lhs->ret_type, BinaryOpStmt::rhs->ret_type, BinaryOpStmt::ret_type` are the same.